### ISSUE TITLE
Refresh does not work for materialized views

### ISSUE THREAD

**Issue by metdos** [citusdata/citus#712]
We allow to create a MATERIALIZED VIEW for some router executable cases, but not to refresh it.

``` sql
CREATE MATERIALIZED VIEW materialized_view_test AS 
SELECT * from lineitem where l_orderkey = 1;
```

``` sql
REFRESH MATERIALIZED VIEW materialized_view_test;
ERROR:  cannot plan queries that include both regular and partitioned relations
```

**Comment by metdos** [comment]
Related to #40.

**Comment by metdos** [comment]
The problem with materialized view creation in real-time and task-tracker queries(#713) is fixed, but the problem with REFRESH continues for all type of executors.

**Comment by anarazel** [comment]
On 2017-03-14 02:42:46 -0700, Metin Döşlü wrote:
> The problem with materialized view creation in real-time and task-tracker queries(#713) is fixed, but the problem with REFRESH continues for all type of executors.

That's just the nextval() though right?  I.e. after merging Marco's PR
removing the jobid, it'll work?

**Comment by metdos** [comment]
> That's just the nextval() though right?  I.e. after merging Marco's PR
> removing the jobid, it'll work?

Actually, I get `ERROR:  cannot plan queries that include both regular and partitioned relations`. I didn't get the connection with `nextval()`.

**Comment by ozgune** [comment]
@metdos -- The underlying query only touches a distributed relation. Do we know why we're emitting this message on a ``REFRESH`` for materialized views? Does it have something to do with how we check the "relations" in a query?

**Comment by anarazel** [comment]
>  I didn't get the connection with nextval().

We used to generate jobids using a sequence.  So every non-router query couldn't be executed in a readonly context.

> Does it have something to do with how we check the "relations" in a query?

I think it's quite possible that it's actually already fixed. We used to not allow view references in queries, but we've recently-ish fixed that.

**Comment by metdos** [comment]
We still have this problem; PostgreSQL adds materialized view rte's to query tree.
```
:rtable (
       {RTE 
       :alias 
          {ALIAS 
          :aliasname old 
          :colnames <>
          }
       :eref 
          {ALIAS 
          :aliasname old 
          :colnames ("l_orderkey" "l_partkey" "l_suppkey" "l_linenumber" "l_qua
          ntity" "l_extendedprice" "l_discount" "l_tax" "l_returnflag" "l_lines
          tatus" "l_shipdate" "l_commitdate" "l_receiptdate" "l_shipinstruct" "
          l_shipmode" "l_comment")
          }
       :rtekind 0 
       :relid 18070 
       :relkind m 
       :tablesample <> 
       :lateral false 
       :inh false 
       :inFromCl false 
       :requiredPerms 0 
       :checkAsUser 10 
       :selectedCols (b)
       :insertedCols (b)
       :updatedCols (b)
       :securityQuals <>
       }
       {RTE 
       :alias 
          {ALIAS 
          :aliasname new 
          :colnames <>
          }
       :eref 
          {ALIAS 
          :aliasname new 
          :colnames ("l_orderkey" "l_partkey" "l_suppkey" "l_linenumber" "l_qua
          ntity" "l_extendedprice" "l_discount" "l_tax" "l_returnflag" "l_lines
          tatus" "l_shipdate" "l_commitdate" "l_receiptdate" "l_shipinstruct" "
          l_shipmode" "l_comment")
          }
       :rtekind 0 
       :relid 18070 
       :relkind m 
       :tablesample <> 
       :lateral false 
       :inh false 
       :inFromCl false 
       :requiredPerms 0 
       :checkAsUser 10 
       :selectedCols (b)
       :insertedCols (b)
       :updatedCols (b)
       :securityQuals <>
      }
.....
```

**Comment by seivan** [comment]
Is this still a problem? How deep do the differences go between Postgres and Citus? I would have expected something like Materialized Views to work out of the box with Citus with claims like "not a fork, but extension"?

**Comment by marcocitus** [comment]
> Is this still a problem? 

This has been fixed on the master branch, and will be in 7.2 with the note that materialized views act as regular postgres tables, as opposed to regular views which act more like distributed tables.

> How deep do the differences go between Postgres and Citus?

Everything you can do in postgres still works after you install Citus. You also have an extra ability which is to turn a table into a distributed table using `create_distribute_table`. Not all postgres commands automatically work with distributed tables, because they may require some special handling that we haven't had a chance to implement yet. `Refresh materialized view` wasn't as high a priority as for example distributed transactions, but now it's been resolved as well.
